### PR TITLE
Create PerilsTableDesktop component

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
- Create PerilsTableDesktop component
- Refactor PerilsTable, create object with variants perils and missing perils to be used in child components

https://github.com/user-attachments/assets/b92eab91-4421-4a2a-96a6-df19b35106f0

## Justify why they are needed
New PerilsTable block

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4673?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4673** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4673?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4671** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4668** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/Perils/Perils.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+'use client'
```
Comment (gustaveen) [inline apps/store/src/components/Perils/Perils.tsx:1]: Next.js warned that we import this component that has `useState`

File: apps/store/src/components/Perils/PerilsTable/PerilsTable.tsx [hunk 2]
```diff
@@ -1,10 +1,50 @@
 'use client'
```
Comment (alebedev) [inline apps/store/src/components/Perils/PerilsTable/PerilsTable.tsx:1]: General question - do we expect to use this component only for products that have multiple tiers? Do others get old perils block if so?

File: apps/store/src/components/Perils/PerilsTable/PerilsTable.tsx [hunk 3]
```diff
@@ -1,10 +1,50 @@
...
-  return <div>{variant === 'mobile' && <PerilsTabs variants={variants} />}</div>
+  const { variants: productVariants } = useProductData()
+
+  const allPerils = useMemo(() => {
+    const perilsMap = new Map(
+      productVariants.flatMap((productVariant) =>
+        productVariant.perils.map((peril) => [peril.title, peril]),
+      ),
+    )
+    return Array.from(perilsMap.values())
```
Comment (alebedev) [inline apps/store/src/components/Perils/PerilsTable/PerilsTable.tsx:19]: Did you consider using `Map` without converting to array?

**Review by alebedev — APPROVED** [review]
Looks good

**Review by alebedev — APPROVED** [review]

**Comment by gustaveen** [comment]
### Merge activity

* **Sep 5, 5:06 AM EDT**: @gustaveen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4673).

**Pull request merged**